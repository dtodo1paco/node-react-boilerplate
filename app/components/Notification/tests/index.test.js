import React from 'react';

import { IntlProvider, defineMessages } from 'react-intl';
import TestHelper from 'utils/testHelper';
import TestedComponent, { Notification } from '../index';
import { createRender, getClasses } from '@material-ui/core/test-utils';

// global vars

const variants = ['success', 'warning', 'error', 'info'];
const dataError = {id:'403',type:'error',text:'error.server.down'};
const dataSuccess = {id:'200',type:'success',text:'success.saved'};

const onClose = jest.fn();

const createComponent = (variant, data, open = true) => <Notification
    classes={{}}
    open={open}
    variant={variant}
    data={data}
    onClose={onClose}
/>

describe('<Notification />', () => {
  describe('snapshot', () => {
    it('should match the snapshot', () => {
      const wrappedComp = TestHelper.wrapComponent(createComponent('error', dataError));
      const tree = TestHelper.render(wrappedComp);
      expect(tree).toMatchSnapshot();
    });
  });
  describe('render', () => {
    it('should render nothing when closed', () => {
      const wrappedComp = TestHelper.wrapComponent(createComponent('error',dataError, false));
      const comp = TestHelper.render(wrappedComp);
      expect(comp.find('div').length).toBe(0);
    });
    it('should render a close button', () => {
      const wrappedComp = TestHelper.wrapComponent(createComponent('error',dataError));
      const comp = TestHelper.render(wrappedComp);
      expect(comp.find('div').length).not.toBe(0);
      expect(comp.find('button').length).toBe(1);
    });
  });
  describe('behavoir', () => {
    it('should call onClose when clicking on close button', () => {
      const wrappedComp = TestHelper.wrapComponent(createComponent('error',dataError));
      const comp = TestHelper.mount(wrappedComp);
      const closeButton = comp.find('button').first();
      closeButton.simulate('click', { preventDefault() {} });
      expect(onClose.mock.calls.length).toBe(1);
    });
    it('should autoClose when variant is not error', () => {
      const wrappedComp = TestHelper.wrapComponent(createComponent('success',dataSuccess));
      // const wrappedComp = createComponent('success',dataSuccess);
      const comp = TestHelper.mount(wrappedComp);
      expect(comp.find('button').length).toBe(1);
      const callback = () => {
        console.log("testing...");
        expect(comp.find('button').length).toBe(0);
      }
      TestHelper.timer(callback, 8000);


      /*
            setTimeout(() => {
              console.log("checking state now");
              expect(component.state().open).toEqual(false);
            }, 7000);
      */
    });
  });

});
