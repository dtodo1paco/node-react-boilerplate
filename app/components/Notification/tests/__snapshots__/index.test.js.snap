// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> snapshot should match the snapshot 1`] = `
<div>
  <div
    class="MuiSnackbar-root-1 MuiSnackbar-anchorOriginBottomCenter-3"
  >
    <div
      aria-describedby="message-id"
      class="MuiTypography-root-38 MuiTypography-body1-47 MuiPaper-root-11 MuiPaper-elevation6-19 MuiSnackbarContent-root-8"
      direction="up"
      role="alertdialog"
    >
      <div
        class="MuiSnackbarContent-message-9"
      >
        <span
          id="message-id"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-63"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
          <span>
            There are some problems connecting with server. Try again in a minute.
          </span>
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action-10"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root-76 MuiIconButton-root-70"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-75"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-63"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root-79"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
